latest/docs-ref-conceptual/azure-cli-extensions.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Azure CLI 2.0 Extensions
+description: Using extensions with the Azure CLI 2.0
+keywords: Azure CLI, Extensions
+author: sptramer
+ms.author: sttramer
+manager: routlaw
+ms.date: 10/30/2017
+ms.topic: article
+ms.prod: azure
+ms.technology: azure
+ms.devlang: azurecli
+ms.service: multiple
+---
+
+# Using extensions with the Azure CLI 2.0
+
+Extensions are individual modules not shipped with the Azure CLI itself that allow you to add functionality through new commands. These might be experimental or pre-release offerings, specialized tools that Microsoft has for your needs, or even extensions that you yourself have written. Extensions allow for a degree of flexibility with the CLI that let you modify it to your own needs, without having to ship a lot of additional packages that aren't considered part of the core feature set.
+
+This article will help you understand how to install, update, and remove extensions for the CLI. It should also answer common questions about extension behavior.
+
+## Finding extensions
+
+In order to know what extensions are available, you can use `az extension list-available`. This command lists the available, official extensions which are provided and supported by Microsoft.
+
+## Installing extensions
+
+Once you have found an extension to install, use `az extension add` to get it. If the extension is an official Microsoft extension listed in `az extension list-available`, you can install the extension
+by name.
+
+```azurecli
+az extension add --name <extension-name>
+```
+
+If the extension is from an external resource or you have a direct link to it, you can provide the source URL or local path. This _must_ be a compiled Python wheel file.
+download.
+
+```azurecli
+az extension add --source <URL-or-path>
+```
+
+Once an extension is installed, it can be found under the value of the `$AZURE_EXTENSION_DIR` shell variable. If this variable is unset, by default the value is `$HOME/.azure/cliextensions` on Linux and macOS, and `%USERPROFILE%\.azure\cliextensions` on Windows.
+
+## Updating extensions
+
+Extensions can only be updated by name:
+
+```azurecli
+az extension update --name <extension-name>
+```
+
+If an extension name cannot be resolved by the CLI for whatever reason, reinstall the extension to update it. There is also the possibility that the extension was moved out of preview and became a built-in command for the CLI. You should always update the CLI before updating extensions, although there is not a guarantee extensions will continue to work with later CLI versions.

```

Comment:
I wouldn't want to enforce this statement "You should always update the CLI before updating extensions".
This makes it sound like it is a MUST.